packages/doc/docs/api-building/sql-syntax.mdx

```diff
@@ -150,6 +150,26 @@ from app_data.payments
 
 VulcanSQL will provide the feature by enhancing the [nunjucks](https://mozilla.github.io/nunjucks/templating.html#tags) macro function by importing a macro function from other files in the next version 😃
 
+## Useing HTTP Request data in SQL file
+
+You can easily use `{{ context.req.xxx }}` in the sql file to obtain data from the http request or a custom header as a query condition.
+
+the syntax like the below:
+
+```sql
+select
+  *
+from
+  "artists"
+where
+  {{ context.req.method }} = "GET"
+and
+  {{ context.header.customParameters }} = "custom parameters"
+  ......
+```
+
+For more available parameters, please refer to: [Koajs#Request](https://koajs.com/#request)
```

Comment:
Another suggestion is that a sample for using `sample` in the api schema file, because if user would like to make the build time work with sample, he should also add the `req` to sample:

```sql
-- We use the req in sql file
select 
*,
{{ context.req.method }} as "req-method",
{{ context.req.url }} as "req-url"
from "artists"
where 
ConstituentID = {{ context.params.id }}
and
 {{ context.req.method }} = 'GET'

```

```ts
# API Schema
urlPath: /artist/:id
request:
  - fieldName: id
    fieldIn: path
    description: constituent id
    validators:
      - required
sample:
  req:
    method: GET
    url: 'localhost:3000'
  parameters:
    id: '1'
  profile: duck
profile: duck

``

Reply:
Hi, @kokokuo , I found that there is already a section about [set-sampler](https://vulcansql.com/docs/api-building/api-document#set-sampler) in the document. Maybe we can update this section to inform the user on how to properly configure it.

Reply:
Could you open an issue to `record` the part ?

Reply:
Thanks @JSYOU adding the issue with sample, I record at here https://github.com/Canner/vulcan-sql/issues/146